lib/msf/core/web_services/framework_extension.rb

```diff
@@ -35,50 +39,41 @@ def self.registered(app)
       app.set :data_service_api_token, ENV.fetch('MSF_WS_DATA_SERVICE_API_TOKEN', nil)
       app.set :data_service_cert, ENV.fetch('MSF_WS_DATA_SERVICE_CERT', nil)
       app.set :data_service_skip_verify, to_bool(ENV.fetch('MSF_WS_DATA_SERVICE_SKIP_VERIFY', false))
+
       @@framework = nil
       # Create simplified instance of the framework
-      app.set :framework, Proc.new {
+      app.set :framework, (proc {
         @@framework ||= begin
           init_framework_opts = {
-            'Logger' => ENV.fetch('MSF_WS_DATA_SERVICE_LOGGER', nil)
+            'Logger' => ENV.fetch('MSF_WS_DATA_SERVICE_LOGGER', nil),
+            # SkipDatabaseInit false is the default behavior, however for explicitness - note that framework first
+            # connects to a local database as a pre-requisite to connecting to a remote service to correctly
+            # configure active record
+            'SkipDatabaseInit' => false
           }
           framework = Msf::Simple::Framework.create(init_framework_opts)
-
-          if !app.settings.data_service_url.nil? && !app.settings.data_service_url.empty?
-            framework_db_connect_http_data_service(framework: framework,
```

Comment:
This was broken, lazily loading framework via a proc meant that the value of `self` in this context was now the sinatra app - rather than `Msf::WebServices::FrameworkExtension` as part of the initial extension registration